salt/pillar/nodegroups.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+'''
+=================
+Nodegroups Pillar
+=================
+
+Introspection: to which nodegroups does my minion belong?
+Provides a pillar with the default name of `nodegroups`
+which contains a list of nodegroups which match for a given minion.
+
+.. versionadded:: Carbon
+
+Command Line
+------------
+
+.. code-block:: bash
+    salt-call pillar.get nodegroups
+    local:
+        - class_infra
+        - colo_sj
+        - state_active
+        - country_US
+        - type_saltmaster
+
+Configuring Nodegroups Pillar
+-----------------------------
+
+.. code-block:: yaml
+
+    extension_modules: /srv/salt/ext
+    ext_pillar:
+      - nodegroups:
+          pillar_name: 'nodegroups'
+
+'''
+
+# Import futures
+from __future__ import absolute_import
+
+# Import Salt libs
+from salt.minion import Matcher
+
+# Import 3rd-party libs
+import salt.ext.six as six
+
+
+__author__ = 'Andrew Hammond <andrew.george.hammond@gmail.com>'
+__copyright__ = 'Copyright (c) 2016 AnchorFree Inc.'
+__license__ = 'Apache License, Version 2.0'
```

Comment:
We'd prefer that instead of these three lines, you just add your name to the AUTHORS file.